jujugui/static/gui/src/app/init.js

```diff
@@ -0,0 +1,1590 @@
+
+const ComponentRenderersMixin = require('./init/component-renderers-mixin');
+const DeployerMixin = require('./init/deployer-mixin');
+
+const yui = window.yui;
+
+class GUIApp {
+  constructor(config) {
+    /**
+      The supplied application configuration object on instantiation. Stored
+      here to avoid having to pass it around.
+      @type {Object}
+    */
+    this.applicationConfig = config;
+    /**
+      Holds the modelUUID the application should connect to. It's possible that
+      this value and the value in the modelAPI instance will differ as another
+      portion of the application has set this value prior to switching models.
+      @type {String}
+    */
+    this.modelUUID = config.jujuEnvUUID;
+    /**
+      The default web page title.
+      @type {String}
+    */
+    this.defaultPageTitle = 'Juju GUI';
+    /**
+      The object used for storing a mapping of previously visited user paths
+      to the type of entity (model, store). e.g. /u/spinach/ghost would map to
+      store.
+      @type {Map}
+    */
+    this.userPaths = new Map();
+    /**
+      Track anonymous mode. This value will be set to true when anonymous
+      navigation is allowed, in essence when a GISF anonymous user is being
+      modeling on a new canvas.
+      @type {Boolean}
+    */
+    this.anonymousMode = false;
+    /**
+      Reference to the changesUtils utilities.
+      @type {Object}
+    */
+    this.changesUtils = window.juju.utils.ChangesUtils;
+    /**
+      Reference to the relationUtils utilities.
+      @type {Object}
+    */
+    this.relationUtils = window.juju.utils.RelationUtils;
+    /**
+      Stores the custom event handlers for the application.
+      @type {Object}
+    */
+    this._domEventHandlers = {};
+    /**
+      Timer for the drag over handler.
+      @type {Integer}
+    */
+    this._dragLeaveTimer = null;
+    /**
+      Reference to the juju.Cookies instance.
+      @type {juju.Cookies}
+    */
+    this.cookieHandler = null;
+    /**
+      The keydown event listener from the hotkey activation.
+      @type {Object}
+    */
+    this._hotkeyListener = hotkeys.activate(this);
+    /**
+      The application database
+      @type {Object}
+    */
+    this.db = new yui.juju.models.Database();
+    /**
+      Timer for debouncing database change events.
+      @type {Integer}
+    */
+    this.dbChangedTimer = null;
+    this._domEventHandlers['boundOnDatabaseChanged'] =
+      this.onDatabaseChanged.bind(this);
+    document.addEventListener(
+      'update', this._domEventHandlers['boundOnDatabaseChanged']);
+    /**
+      Application instance of the user class.
+      @type {Object}
+    */
+    this.user = config.user || new window.jujugui.User({
+      externalAuth: config.auth,
+      expiration: window.sessionStorage.getItem('expirationDatetime')
+    });
+    /**
+      Stores the user object for the charmstore.
+      @type {Object}
+    */
+    this.users = csUser.create();
```

Comment:
I agree on all accounts and have actually already discussed doing just that with JC. However (I'm sure saying that a lot) I feel that it's out of scope of this branch as this was just the first step to move the code over (and then some...) So I'll make a note to do this in follow-ups